### ISSUE TITLE
Sagas finishing too early

### ISSUE THREAD

**Issue by Cretezy** [bmealhouse/next-redux-saga#57]
I have to following setup:

- In `getInitialProps`, I dispatch `fetch`
- A saga listens for the `fetch` action, and runs (this works)
- Once this saga is done, it dispatches another action `fetched` (this works)
- Another saga listen for the `fetched` action (this doesn't work)

It seems like it's finishing too early, and not letting the second action be ran in the sagas.

How could I fix this? Is this my application code?

**Comment by Cretezy** [comment]
I currently fixed this by calling the saga directly instead of using an action, although I feel like all actions should be handled before finishing

**Comment by bbortt** [comment]
Hi @Cretezy
can you please post the root saga?
Br, Timon

**Comment by Cretezy** [comment]
Roughly:

```js
function* rootSaga() {
  yield all([
    takeEvery(ACTION1, handleAction1),
    takeEvery(ACTION2, handleAction2),
  ]);
}

function* handleAction1() {
  console.log("Handling 1");
  yield put({ type: ACTION2 });
}

function* handleAction2() {
  console.log("Handling 2");
}
```
```js
Page.getInitialProps({ store }) {
  store.dispatch({ type: ACTION1 });
}
```

Actual Result:
```
Handling 1
```
Expected Result:
```
Handling 1
Handling 2
```

**Comment by bbortt** [comment]
The `yield all([ takeEvery, takeEvery])` will exit on first saga finished. You need to start an async thread e.g. using `yield all([ fork, fork])` or `spawn` instead of `fork`. This will wait until all threads return. You'll find it in the [saga docs](https://redux-saga.js.org/docs/advanced/RootSaga.html). 
Br, Timon

**Comment by Cretezy** [comment]
Let me try this and get back to you, thank you!

**Comment by bbortt** [comment]
Here's an [example](https://github.com/bbortt/qdrakeboo/tree/master/source/frontend/next-js-web-ui/app/state/saga).

**Comment by bbortt** [comment]
@Cretezy did it work? Would be great to be able to close this issue 🚀

**Comment by Cretezy** [comment]
Haven't had time to test this since I reverted to calling the saga itself.

**Comment by bbortt** [comment]
Think this might be solved in `v4.1.1`. Closing due to inactivity.
